### ISSUE TITLE
ExceptionWrapper errors

### ISSUE THREAD

**Issue by entering** [FriendsOfSymfony/FOSRestBundle#677]
FOS\RestBundle\Util\ExceptionWrapper

```php
    public function __construct($data)
    {
        $this->code = $data['status_code'];
        $this->message = $data['message'];

        if (isset($data['errors'])) {
            $this->errors = $data['errors'];
        }
    }
```

When an invalid form is return the errors are displayed there, but would be really nice to be able to throw exceptions and if they implement some interface (like getErrors()) or have defined in rest configuration they would appear on the errors "field".

I'm not sure this is possible because the ExceptionWrapper is created on ExceptionController and this code:

```php
        if (!$exception instanceof DebugFlattenException && !$exception instanceof HttpFlattenException) {
```

The original exception is transformed by symfony into a DebugFlattenException, even if the exception have a property errors and a getErrors that data will be lost. I had a look on FlattenException and there is no way to get the original exception from there.

This makes sense? Maybe this can be solved if FlattenException can keep the original exception inside.

**Comment by magnetik** [comment]
I don't really understand how this PR helps to solve the bug described here. 

Should I use custom serialization groups ?

**Comment by Ener-Getick** [comment]
Fixed by #1358
